ambry-tools/src/main/java/com.github.ambry/clustermap/HelixVcrPopulateTool.java

```diff
@@ -0,0 +1,190 @@
+/*
+ * Copyright 2019 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ */
+package com.github.ambry.clustermap;
+
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import joptsimple.ArgumentAcceptingOptionSpec;
+import joptsimple.OptionParser;
+import joptsimple.OptionSet;
+import joptsimple.OptionSpec;
+import org.apache.helix.ConfigAccessor;
+import org.apache.helix.HelixAdmin;
+import org.apache.helix.controller.rebalancer.strategy.CrushEdRebalanceStrategy;
+import org.apache.helix.manager.zk.ZKHelixAdmin;
+import org.apache.helix.manager.zk.ZKHelixManager;
+import org.apache.helix.manager.zk.ZNRecordSerializer;
+import org.apache.helix.manager.zk.client.DedicatedZkClientFactory;
+import org.apache.helix.manager.zk.client.HelixZkClient;
+import org.apache.helix.model.ClusterConfig;
+import org.apache.helix.model.HelixConfigScope;
+import org.apache.helix.model.IdealState;
+import org.apache.helix.model.LeaderStandbySMD;
+import org.apache.helix.model.builder.FullAutoModeISBuilder;
+import org.apache.helix.model.builder.HelixConfigScopeBuilder;
+import org.apache.helix.tools.ClusterSetup;
+
+
+/**
+ * This tool provides function to create vcr cluster and update vcr cluster by referencing src cluster.
+ */
+public class HelixVcrPopulateTool {
+
+  private static String SEPARATOR = "/";
+  static List<String> ignoreResourceKeyWords = Arrays.asList("aggregation", "trigger", "stats");
+
+  public static void main(String[] args) {
+    OptionParser parser = new OptionParser();
+    OptionSpec createClusterOpt = parser.accepts("createCluster",
+        "Create resources in dest cluster. --createCluster --dest destZkEndpoint/destClusterName");
```

Comment:
Does it actually create resources or just the cluster?

Reply:
just cluster.

Reply:
Will add in comment.

Reply:
Issues addressed in https://github.com/linkedin/ambry/pull/1179